Jishi.SonosUPnP/SonosDiscovery.cs

```diff
@@ -71,8 +71,8 @@ private void UpdateTopology(string value)
 			Console.WriteLine(value);
 
 			Zones.Clear();
-
-			foreach (var zoneNode in xml.Elements("ZoneGroup"))
+			foreach (var zonegroup in xml.Elements("ZoneGroups"))
+			foreach (var zoneNode in zonegroup.Elements("ZoneGroup"))

```

Comment:
I don't think the XML has changed, are you sure this is necessary? Doesn't `.Elements()` search more than one level deep?

Reply:
Yes. generation 1 and 2 could only be found with the change.